azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub.py

```diff
@@ -35,11 +35,16 @@ def __init__(self):
     @pipeline_thread.runs_on_pipeline_thread
     def _run_op(self, op):
         if isinstance(op, pipeline_ops_base.EnableFeatureOperation):
-            # If we're enabling twin patches, we set last_version_seen to -1
-            # as a way of enabling this functionality.  If the ConnectedEvent handler
-            # sees this -1, it will send a GetTwinOperation to refresh desired properties.
-
-            if op.feature_name == constant.TWIN_PATCHES:
+            # Ensure_desired_properties enables twin patches, when true, by setting last version
+            # to -1. The ConnectedEvent handler sees this and sends a GetTwinOperation to refresh
+            # desired properties. Setting ensure_desired_properties to false causes the GetTwinOp
+            # to not be sent. The rest of the functions in this stage stem from the GetTwinOperation,
+            # so disabling ensure_desired_properties effectively disables this stage.
+
+            if (
```

Comment:
I would put this conditional even farther out - wrap the entire content of _run_op() in this condition